### PR TITLE
Add API for renaming Worlds

### PR DESCRIPTION
**Pull request by FerusGrim**
(also remove unused java.io,File import)

### CODE REVIEW

**Comment by modwizcode** [comment]
Can you amend this to fix the deleteWorld javadoc?

**Comment by modwizcode** [comment]
Please update to follow _blood's comments from IRC

**Comment by ST-DDT** [comment]
Can we somehow change the PermissionContext of a world. So that if we have renamed the world the permission service still detects this world as the original one?

**Comment by FerusGrim** [comment]
@ST-DDT Correct me if I'm wrong, but from what I can tell this would be done by the implementation.

**Comment by ST-DDT** [comment]
Currently the world context is directly connected to the worlds name. But i would like to set it myself using API methods. It does not really have anything in common with renaming a world besides that both are associated with worlds and changing the identifier.

Just for the reference
https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/service/permission/context/Contextual.java

**Comment by ryantheleach** [comment]
Really the name should just be a friendly name for users to use. If worlds have a UUID shouldn't the permissions service associate with the worlds UUID, and only use the name for display purposes?

**Comment by modwizcode** [comment]
Rename will preserve UUID

**Comment by ST-DDT** [comment]
@ryantheleach Please rememer the permission service must store the value to the config, and it should be human readable at that point.
I'm asking for this feature, because i would like to create some temporary worlds (for games) that are connected to a specific permission context, without having to somehow inject it to all joined players.

**Comment by modwizcode** [comment]
Because this issue could take awhile to work out, I'm going to split the offline import functionality into a separate PR that can be merged quicker.

**Comment by modwizcode** [comment]
Merged into PR commit https://github.com/SpongePowered/SpongeAPI/commit/142f2eaf011fb4e0972b8bcea9e6922232632a22.

File: src/main/java/org/spongepowered/api/Server.java [hunk 1]
```diff
@@ -206,6 +206,49 @@
...
+     */
+    Optional<WorldProperties> renameWorld(WorldProperties worldProperties, String newName);
+
+    /**
+     * Deletes the provided world's files from the disk.
+     *
+     * @param worldProperties The world instance to delete
+     * @return True if the deletion was successful.
+     */
+    boolean deleteWorld(WorldProperties worldProperties);
```
Comment (spaceemotion) [inline src/main/java/org/spongepowered/api/Server.java]: Will this only delete the world's `level.dat` file or the whole folder that contains the region files as well?

Comment (JBYoshi) [inline src/main/java/org/spongepowered/api/Server.java]: The whole folder.

**Comment by gabizou** [comment]
@modwizcode The merge commit has only one method from this PR, is this intended? Why were the others left out?

**Comment by modwizcode** [comment]
there was only one? it's a merge into my PR branch?

**Comment by JBYoshi** [comment]
They were added to modwizcode's PR in other commits.

**Pull request closed without merging**